### ISSUE TITLE
Issues with Version: 1.10.4-0

### ISSUE THREAD

**Issue by jdeon-classy** [duosecurity/duo_unix#137]
When changing from version 1.10.1-0 to version Version: 1.10.4-0 we can no longer use duo-unix as mfa when sshing into our servers. As a temporary fix we have locked out duo version to 1.10.1-0 and everything works as expected. I am not linking our sshd config, but we have everything identical to the instructions found [here](https://duo.com/docs/duounix) for Ubuntu 16.

OS:
Ubuntu 16.04.5 LTS (GNU/Linux 4.4.0-1067-aws x86_64)

Version:
username@ip-<ip-address>:~$ dpkg -s duo-unix | grep Version
Version: 1.10.4-0

/var/log/auth.log:
```
Sep 12 21:49:41 ip-<ip-address> sshd[27406]: Connection from <ip-address> port <port> on server-ip port 22
Sep 12 21:49:41 ip-<ip-address> sshd[27406]: Failed publickey for username from <ip-address> port <port> ssh2: RSA SHA256:<sha-value>
Sep 12 21:49:41 ip-<ip-address> sshd[27406]: Postponed publickey for username from <ip-address> port <port> ssh2 [preauth]
Sep 12 21:49:41 ip-<ip-address> sshd[27406]: Partial publickey for username from <ip-address> port <port> ssh2: RSA SHA256:<sha-value>
Sep 12 21:49:41 ip-<ip-address> sshd[27406]: Failed keyboard-interactive/pam for username from <ip-address> port <port> ssh2
Sep 12 21:49:41 ip-<ip-address> sshd[27406]: Connection closed by <ip-address> port <port> [preauth]
```

ssh output:
```
ssh -vv -i ~/.ssh/pubkey username@<server-ip>
OpenSSH_7.4p1, LibreSSL 2.5.0
debug1: Reading configuration data /Users/username/.ssh/config
debug1: Reading configuration data /etc/ssh/ssh_config
debug2: resolving "<server-ip>" port 22
debug2: ssh_connect_direct: needpriv 0
debug1: Connecting to <server-ip> [<server-ip>] port 22.
debug1: Connection established.
debug1: identity file /Users/username/.ssh/pubkey type 1
debug1: key_load_public: No such file or directory
debug1: identity file /Users/username/.ssh/pubkey-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_7.4
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.2p2 Ubuntu-4ubuntu2.4
debug1: match: OpenSSH_7.2p2 Ubuntu-4ubuntu2.4 pat OpenSSH* compat 0x04000000
debug2: fd 3 setting O_NONBLOCK
debug1: Authenticating to <server-ip>:22 as 'username'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug2: local client KEXINIT proposal
debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,ext-info-c
debug2: host key algorithms: ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa
debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc
debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc
debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
debug2: compression ctos: none,zlib@openssh.com,zlib
debug2: compression stoc: none,zlib@openssh.com,zlib
debug2: languages ctos: 
debug2: languages stoc: 
debug2: first_kex_follows 0 
debug2: reserved 0 
debug2: peer server KEXINIT proposal
debug2: KEX algorithms: curve25519-sha256@libssh.org,ecdh-sha2-nistp384
debug2: host key algorithms: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
debug2: ciphers ctos: aes256-gcm@openssh.com,aes256-ctr,chacha20-poly1305@openssh.com
debug2: ciphers stoc: aes256-gcm@openssh.com,aes256-ctr,chacha20-poly1305@openssh.com
debug2: MACs ctos: hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,umac-128-etm@openssh.com
debug2: MACs stoc: hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,umac-128-etm@openssh.com
debug2: compression ctos: none,zlib@openssh.com
debug2: compression stoc: none,zlib@openssh.com
debug2: languages ctos: 
debug2: languages stoc: 
debug2: first_kex_follows 0 
debug2: reserved 0 
debug1: kex: algorithm: curve25519-sha256@libssh.org
debug1: kex: host key algorithm: ecdsa-sha2-nistp256
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:<sha-value>
debug1: Host '<server-ip>' is known and matches the ECDSA host key.
debug1: Found key in /Users/username/.ssh/known_hosts:245
debug2: set_newkeys: mode 1
debug1: rekey after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug2: set_newkeys: mode 0
debug1: rekey after 134217728 blocks
debug2: key: /Users/username/.ssh/id_rsa_name (0x7f9b4fd00ab0), agent
debug2: key: /Users/username/.ssh/pubkey (0x7f9b4fd019b0), explicit
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<rsa-sha2-256,rsa-sha2-512>
debug2: service_accept: ssh-userauth
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey
debug1: Next authentication method: publickey
debug1: Offering RSA public key: /Users/username/.ssh/id_rsa_name
debug2: we sent a publickey packet, wait for reply
debug1: Authentications that can continue: publickey
debug1: Offering RSA public key: /Users/username/.ssh/pubkey
debug2: we sent a publickey packet, wait for reply
debug1: Server accepts key: pkalg rsa-sha2-512 blen 535
debug2: input_userauth_pk_ok: fp SHA256:<sha-value>
Authenticated with partial success.
debug1: Authentications that can continue: keyboard-interactive
debug1: Next authentication method: keyboard-interactive
debug2: userauth_kbdint
debug2: we sent a keyboard-interactive packet, wait for reply
debug1: Authentications that can continue: keyboard-interactive
debug2: we did not send a packet, disable method
debug1: No more authentication methods to try.
Permission denied (keyboard-interactive).
```

**Comment by xdesai** [comment]
Hi!
After looking a little further I believe there's a good chance that your issue is related to the bug we just fixed in 1.10.5. Can you please update and see if this resolves things for you?

**Comment by xdesai** [comment]
Closing out now. Please let us know if you are still experiencing issues.
